purple/settings/base.py

```diff
@@ -157,3 +157,17 @@
 CELERY_BROKER_URL = os.environ.get("PURPLE_BROKER_URL", "amqp://mq/")
 CELERY_BROKER_CONNECTION_RETRY_ON_STARTUP = True
 CELERY_TASK_IGNORE_RESULT = True  # ignore results unless specifically enabled
+
+# Crossref / DOI
+CROSSREF_API = "https://test.crossref.org/servlet/deposit"
+CROSSREF_USER = "user"
+CROSSREF_PASSWORD = "friend"
```

Comment:
I assume the api/user/password need to become a secret for production? If so, these should be pulled from the environment as `PURPLE_CROSSREF_API_URL`, `PURPLE_CROSSREF_USER` and `PURPLE_CROSSREF_PASSWORD`. We've been trying to do most of this config in `base.py` by using the environment if available and falling back to dev defaults, but it could go in `production.py` if the logic is different.

If `PURPLE_CROSSREF_API_URL` is set in the environment, then the absence of either of the other two settings should be an error. The production config should probably check that the API_URL is set. We'll need to be careful not to give staging the real secrets.

Reply:
f3d11b34e2280ec73264594f0982c36ce49c507d introduces ENV lookup.